modules/codegen/src/smithy4s/codegen/Codegen.scala

```diff
@@ -36,34 +37,42 @@ object Codegen { self =>
       args.localJars
     )
 
-    val scalaFiles = if (!args.skipScala) {
-      Codegen.generate(model, args.allowedNS, args.excludedNS).map {
-        case (relPath, name, outputString) =>
-          val fileName = name + ".scala"
-          val scalaFile = (args.output / relPath / fileName)
-          os.write.over(scalaFile, outputString, createFolders = true)
-          scalaFile
+    val (scalaFiles, smithyResources) = if (!args.skipScala) {
```

Comment:
should we read:

```
val scalaFiles = if (args.skipScala) { Nil } else { .. }
val openApiFiles = if (args.skipOpenapi) { Nil } else { .. }
val resources = if (args.skipResources) { Nil } else { .. }
```

What I mean, is there, it the current setup `--skip scala` will not produce resources files, which seems counter-intuitive. Maybe there should not be a way to skip `resources. They would be skipped if you `--skip scala`

Reply:
Nah, the resources in question are MEANT to be tied to the presence of Scala code, in particular because one of them literally exists to say "the Scala code for these particular namespaces was already generated, you can skip it downstream". People who want to package resources without Scala code should simply disable the Smithy4s plugin and manually create the resources, like we do in the protocol module. 

These options will be documented anyway, and I don't expect users to tweak them often.